api/rocdecode.h

```diff
@@ -1500,6 +1500,13 @@ extern rocDecStatus ROCDECAPI rocDecGetVideoFrame(rocDecDecoderHandle decoder_ha
                                                     void *dev_mem_ptr[3], uint32_t (&horizontal_pitch)[3],
                                                     RocdecProcParams *vid_postproc_params);
 
+/************************************************************************************************************************/
+//! \fn extern rocDecStatus ROCDECAPI rocDecReleaseVideoFrame(rocDecDecoderHandle decoder_handle, void *parser_handle, int pic_idx)
+//! \ingroup group_amd_rocdecode
+//! Release video frame corresponding to pic_idx for reuse after consuming the interopped device memory
+/************************************************************************************************************************/
+extern rocDecStatus ROCDECAPI rocDecReleaseVideoFrame(rocDecDecoderHandle decoder_handle, void* parser_handle, int pic_idx);
```

Comment:
Our decoder and parser APIs are two separate entities. Therefore, passing the `void* parser_handle` to a decoder API (i.e., `rocDecReleaseVideoFrame`) is against the flow.

Besides that, you also define the `rocDecParserReleaseFrame` API in the `rocparser.h` which should be enough for our purpose of releasing a frame associated with `pic_idx`.

Essentially, we can call the `rocDecParserReleaseFrame` API from the [`RocVideoDecoder::ReleaseFrame`](https://github.com/ROCm/rocDecode/blob/develop/utils/rocvideodecode/roc_video_dec.cpp#L881) function like  
`ROCDEC_API_CALL(rocDecParserReleaseFrame(rocdec_parser_, fb->picture_index));` after [line 888](https://github.com/ROCm/rocDecode/blob/develop/utils/rocvideodecode/roc_video_dec.cpp#L888)
to release the frame associated with `fb->picture_index` at parser level. Using this method we don't need to add 
`rocDecReleaseVideoFrame ` API at the decoder level.

Reply:
I have discussed this yesterday. We have rocDecGetVideoFrame() on rocDecoder. That was the reason for having the ReleaseVideoFrame(). Since it has to be connected to parser, I had to pass the parser handle.  We need to carefully analyze if there is anything needed to be sone in decoder when we release the frame for reuse. We can discuss this